File: java/complete/myretro/src/main/java/com/apress/myretro/user/UserController.java
package com.apress.myretro.user;


import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@AllArgsConstructor
@RestController
@RequestMapping("/users")
public class UserController {

    private UserRepository userRepository;

    @GetMapping
    Iterable<User> getAllUsers() {
        return userRepository.findAll();
    }

    @GetMapping("/{email}")
    User getUserByEmail(@PathVariable String email){
        return userRepository.findById(email).get();
    }

    @PostMapping
    User addNewUser(@RequestBody User user){
        userRepository.save(user);
        return user;
    }

    @DeleteMapping("/{email}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    void deleteUser(@PathVariable String email){
        userRepository.delete(email);
    }
}

File: kotlin/complete/myretro/src/main/kotlin/com/apress/myretro/user/UserController.kt
package com.apress.myretro.user

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController (val userRepository: UserRepository) {

    @GetMapping
    fun getAllUsers():Iterable<User> = userRepository.findAll()

    @GetMapping("/{email}")
    fun getUserByEmail(@PathVariable email:String): User? = userRepository.findById(email)

    @PostMapping
    fun addNewUser(@RequestBody user: User): User = userRepository.save(user)

    @DeleteMapping("/{email}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteUser(@PathVariable email:String) = userRepository.delete(email)
}